src/devnet/integrated-devnet.ts

```diff
@@ -0,0 +1,41 @@
+import { ChildProcess } from "child_process";
+
+export abstract class IntegratedDevnet {
+    protected childProcess: ChildProcess;
+
+    constructor(protected host: string, protected port: string) {
+        IntegratedDevnet.cleanupFns.push(this.cleanup.bind(this));
+    }
+
+    protected static cleanupFns: Array<() => void> = [];
```

Comment:
If exitHook can be made a part of the constructor or start method, maybe there won't be need for having static classes.